Add Vagrantfile for ubuntu

fbs opened the pull request:
I wanted to toy with bpftrace  but as a "mac user" I have to rely on VMs to do so. Copy/pasting the install instructions into a new VM is a bit tedious and can easily be automated using Vagrant, making it as simple as `vagrant up ubuntu-18.04; vagrant ssh ubuntu-18.04`.

This commit adds a very basic Vagrantfile for a few ubuntu versions. It still has some issues and isn't ready to be merged yet (hence the draft) but the basic install on 18.04 & 18.10 works.

Is this something that would be useful for the bpftrace project?

brendangregg commented:
Yes, I know people using Vagrant, so this looks useful to me.

mmarchini commented:
I like the idea of having a Vagrantfile, should be helpful for OS X users as well as users who want to test across Kernel versions. But I think a Vagrantfile in the repo should be focused on development. Not sure what others think though. 

A few suggestions to make it more focused on development:

- Use synced_folder instead of cloning the project
- Don't build bpftrace during provision
- Install libbcc from iovisors ppa instead of building it (should be faster)

fbs commented:
> A few suggestions to make it more focused on development:
>     * Use synced_folder instead of cloning the project

Agreed.

>     * Don't build bpftrace during provision

I took it out and added the build instructions back in as a post-up message. 

>     * Install libbcc from iovisors ppa instead of building it (should be faster)

Oh didn't  know about those ppas, those are useful. I've kept the option build `bcc` in hidden behind an env var (, might be useful for systems on which bcc isn't packaged.

Usage would be: `BUILD_BCC=1 vagrant up ubuntu-18.04`

I've updated the MR. Still need to add some refences in the docs though.

fbs commented:
> * Install libbcc from iovisors ppa instead of building it (should be faster)

I tried this but ran into issues with 18.10 as `bpf.h` was missing. Building bcc from source seems like the easiest way to make sure everything works on all platforms. It also keeps the amount of inline scripts to a minimum which keeps maintenance easier.
The build time isn't that bad either, spinning up 2 VMs at the same time on my mac (i5-8259U CPU @ 2.30GHz, nvme) took:

```
vagrant up ubuntu-18.10  6.48s user 3.03s system 3% cpu 5:05.42 total
vagrant up ubuntu-18.04  7.08s user 3.72s system 3% cpu 5:29.97 total
```

Could maybe win a bit by running more make jobs, but I ran into OOM conditions from time to time.

Rebuilding the VM isn't something you have to do often either.
